agent/api/container_unix_test.go

```diff
@@ -0,0 +1,35 @@
+// +build !windows,!integration
```

Comment:
Why this is not enabled on windows, the `BuildResourceDependency` isn't specific to linux?

Reply:
I am using `CgroupResource` for this test, which is specific to linux. hence added the test here. When we enable a common resource, we can add tests to `container_test.go`.